### ISSUE TITLE
Upgrading to 1.6.1: Error in DDL upgrade 2021081000: [1050] Table 'responses' already exists

### ISSUE THREAD

**Issue by darren12345** [roundcube/roundcubemail#8880]
I received this error when upgrading from 1.6.0:

```
Running update script at target...
Executing database schema update.
ERROR: Error in DDL upgrade 2021081000: [1050] Table 'responses' already exists
Updating database schema (2021081000)... [FAILED]
All done.
```

Is this something to be concerned about?

**Comment by reinob** [comment]
I also upgraded (from 1.5.3) to 1.6.1 and in the log I get
`DB Error: [1] no such table: contacts (SQL Query: SELECT COUNT(c."contact_id") ...`
(once) for each message I view.

Upgrading the database schema fails too:
```
bin/updatedb.sh --dir=SQL --package=roundcube
Updating database schema (2021100300)... [FAILED]
ERROR: Error in DDL upgrade 2021100300: [1] no such table: contacts
```

**Comment by alecpl** [comment]
Are you using `db_prefix` setting? Normally these tables should be in the database, so it does not make sense.

**Comment by reinob** [comment]
No, no db_prefix. For some reason my (sqlite) db had old_contacts but no contacts. Other tables were missing. I went through the statements in SQL/sqlite/202110030.sql in chunks, and had errors when DROPing INDEX ix_*. It seems I didn't have any index.

In any case, now my database looks OK. Usually I've done the updates running bin/installto.sh, but I think that I may have missed a database schema update or two, and now it hit me.

As far as I'm concerned, my issue can be considered to be a user/admin failure. In that sense, sorry for hijacking this issue. Hopefully @darren12345 will provide further feedback about their issue!

Thanks in any case @alecpl  for your reply!

**Comment by darren12345** [comment]
Well they’re different problems in that you were missing tables but my error is because I already have an extant table. I haven’t drilled into the DB (MariaDB) but roundcube does seem to be working OK.

**Comment by alecpl** [comment]
@darren12345 How do you call the update script? What version did you upgrade from?

**Comment by darren12345** [comment]
Running the installto.bin. From 1.6.0.

**Comment by alecpl** [comment]
How? Why from 1.6.0? You should be using the one from the version you upgrade to.

**Comment by darren12345** [comment]
No, you are misunderstanding. 

“How do you call the update script?“
Running the installto.sh.

“What version did you upgrade from?”
From 1.6.0

**Comment by alecpl** [comment]
What is your command look like precisely? What do you have in the `system` table? Could you enable `sql_debug` and provide the log?

**Comment by darren12345** [comment]
The command would have just been ./installto.sh /home/web/mail.

```
MariaDB [roundcube]> select * from system
    -> ;
+-------------------+------------+
| name              | value      |
+-------------------+------------+
| roundcube-version | 2020122900 |
+-------------------+------------+
1 row in set (0.032 sec)
```
Is there any particular action I should take when running the msyql debug that might show any problems, or just general usage of Roundcube?

**Comment by darren12345** [comment]
OK, I'm slightly confused here.

1. the initial upgrade otherwise appeared to work. RC is working fine and is reporting the correct value.
2. I can see in the file mysql.initial.sql file there is a statement `INSERT INTO `system` (`name`, `value`) VALUES ('roundcube-version', '2022081200');` which I assume based on the name is only run on a fresh install. In the SQL/mysql directory are what I presume are mysql files that are run on each subsequent upgrade. None of them, include a statement that updates the version number in the system table, and definitley not the four names after versions later than the one my system is reporting, so where is that value updated during an upgrade?

**Comment by alecpl** [comment]
It's supposed to be updated by the update script automatically if the specific migration succeeded.

**Comment by darren12345** [comment]
OK. I changed the value in the system table as you suggested and ran the installto.sh script again. It looked slightly different this time (it automatically ran a composer update which I don't think it's done before) but importantly there were no errors. System table version now reports:
```
MariaDB [roundcube]> select * from system;
+-------------------+------------+
| name              | value      |
+-------------------+------------+
| roundcube-version | 2022081200 |
+-------------------+------------+
1 row in set (0.003 sec)
```

So I guess that's correct. Thanks for your help.
